ui-core/src/main/java/com/adyen/checkout/components/ui/view/ComponentViewType.kt

```diff
@@ -8,4 +8,11 @@
 
 package com.adyen.checkout.components.ui.view
 
+import com.adyen.checkout.components.ui.R
+
 interface ComponentViewType
+
+interface ButtonComponentViewType : ComponentViewType {
+    val buttonTextResId: Int
```

Comment:
Not really a fan of default values in interfaces. I think it's better if the implementation explicitly says what value it's using. And then if you want to easily re-use the same value I would define a companion object in this interface with `@StringRes const val DEFAULT_BUTTON_TEXT = R.string.pay_button`.

Also can you annotate this field with `@StringRes`?